src.ts/providers/signer-noncemanager.ts

```diff
@@ -73,15 +73,18 @@ export class NonceManager extends AbstractSigner {
     }
 
     async sendTransaction(tx: TransactionRequest): Promise<TransactionResponse> {
-        const noncePromise = this.getNonce("pending");
-        this.increment();
+        try {
+            const noncePromise = this.getNonce("pending");
+            this.increment();
 
-        tx = await this.signer.populateTransaction(tx);
-        tx.nonce = await noncePromise;
+            tx = await this.signer.populateTransaction(tx);
+            tx.nonce = await noncePromise;
 
-        // @TODO: Maybe handle interesting/recoverable errors?
-        // Like don't increment if the tx was certainly not sent
-        return await this.signer.sendTransaction(tx);
+            return await this.signer.sendTransaction(tx);
+        } catch (err) {
+            this.#delta--;
```

Comment:
If the error happens in getNonce, we shouldn't reduce #delta.

Reply:
Ignore previous comment. The noncePromise is awaited after we increment, so we should always decrease upon error.